discovery/azure/azure.go

```diff
@@ -64,6 +66,10 @@ var DefaultSDConfig = SDConfig{
 	AuthenticationMethod: authMethodOAuth,
 }
 
+func init() {
+	config.RegisterServiceDiscovery("azure_sd_configs", &SDConfig{})
```

Comment:
The suffix _sd_configs should be enforced.

Maybe the public RegisterServiceDiscovery could add it and call a non-exported registerServiceDiscovery:

RegisterServiceDiscovery(name){
registerServiceDiscovery ( name+"_sd_configs")
}

and we register the static configs with registerServiceDiscovery ("static_configs")

Reply:
SGTM

Reply:
I would like the signature to be `RegisterServiceDiscovery(config discoverer.Config)` and use the `config.Name()` to set the YAML key name (e.g. `config.Name() + "_sd_configs"`). The only issue with that is there are two discovery mechanisms with logger values that are different from their YAML key names:
- kubernetes
  - yaml: kubernetes_sd_configs
  - log: discovery=k8s
- serverset
  - yaml: serverset_sd_configs
  - log: discovery=zookeeper

Are you okay with changing the logger values to match the YAML keys?
- kubernetes
  - yaml: kubernetes_sd_configs
  - log: discovery=kubernetes
- serverset
  - yaml: serverset_sd_configs
  - log: discovery=serverset

Reply:
I am ok with this. cc @brancz and @brian-brazil to double validate that it's ok.

As per https://prometheus.io/docs/prometheus/latest/stability/ the format of the logs is not considered stable.

Reply:
I went ahead with this change for now. If it's going to be a problem, I can push the log labeling into the Config.NewDiscoverer methods and each implementation can choose its own label that may or may not be its name.

Reply:
perfectly happy with this as well